File: java/PE032.java
public class PE032 {
    public static void main(String[] args) {
        int sum = 0;
        int[] a = new int[10];
        for(int i = 1000; i <= 9999; i++) {
            int j = (int)Math.sqrt(i);
            for(int k = 2; k <= j; k++) {
                if(i%k == 0) {
                    int m;
                    for(int s = 0; s < 10; s++) {
                        a[s] = 0;
                    }
                    int n = 0;
                    m = i;
                    while(m != 0) {
                        a[m%10] ++;
                        n++;
                        m /= 10;
                    }
                    m = k;
                    while(m != 0) {
                        a[m%10]++;
                        n++;
                        m /= 10;
                    }
                    m = i / k;      
                    while(m != 0) {
                        a[m%10] ++;
                        n++;
                        m /= 10;
                    }
                    m = 0;
                    for(int s = 1; s < 10; s++)
                           if(a[s] == 1)
                                m++;
                    if(n == 9 && m == 9 && a[0] == 0) {
                        sum += i;
                        break;
                    }
                }
            }
        }
        System.out.println(sum);
    }
}

File: c/PE032.c
#include <stdio.h>
#include <math.h>

int main() {
    int i, j, k, m, n, sum = 0;
    int a[10];
    for(i = 1000; i <= 9999; i++){
        k = sqrt(i);
        for(j = 2; j <= k+1; j++) {
            if(i%j == 0) {
                for(m = 0; m < 10; m++)
                    a[m] = 0;
                n = 0;
                m = i;
                while(m) {
                    a[m%10] = 1;
                    m /= 10;
                    n++;
                }
                m = j;
                while(m) {
                    a[m%10] = 1;
                    m /= 10;
                    n++;
                }
                m = i/j;
                while(m){
                    a[m%10] = 1;
                    m /= 10;
                    n++;
                }
                for(m = 1; m < 10; m++)
                    if(a[m] == 0)
                        break;
                if(m == 10 && n == 9){
                    sum += i;
                    break;
                }
            }
        }
    }
    printf("%d\n", sum);
    return 0;
}
